lib/avo/concerns/has_field_discovery.rb

```diff
@@ -0,0 +1,183 @@
+      ].freeze
+
+      class_methods do
+        def column_names_mapping
+          @column_names_mapping ||= DEFAULT_COLUMN_NAMES_MAPPING.dup
+                                    .except(*COLUMN_NAMES_TO_IGNORE)
+                                    .merge(Avo.configuration.column_names_mapping || {})
+        end
+
+        def column_types_mapping
+          @column_types_mapping ||= DEFAULT_COLUMN_TYPES_MAPPING.dup
+                                    .merge(Avo.configuration.column_types_mapping || {})
+        end
+      end
+
+      # Returns database columns for the model, excluding ignored columns
+      def model_db_columns
+        @model_db_columns ||= safe_model_class.columns_hash.symbolize_keys.except(*COLUMN_NAMES_TO_IGNORE)
+      end
+
+      # Discovers and configures database columns as fields
+      def discover_columns(only: nil, except: nil, **field_options)
+        @only, @except, @field_options = only, except, field_options
+        return unless safe_model_class.respond_to?(:columns_hash)
+
+        model_db_columns.each do |column_name, column|
+          next unless column_in_scope?(column_name)
+          next if reflections.key?(column_name) || rich_texts.key?("rich_text_#{column_name}")
+
+          field_config = determine_field_config(column_name, column)
+          next unless field_config
+
+          field_options = build_field_options(field_config, column)
+          field column_name, **field_options, **@field_options
+        end
+      end
+
+      # Discovers and configures associations as fields
+      def discover_associations(only: nil, except: nil, **field_options)
+        @only, @except, @field_options = only, except, field_options
+        return unless safe_model_class.respond_to?(:reflections)
+
+        discover_by_type(tags, :tags) { |name| name.split("_").pop.join("_").pluralize }
+        discover_by_type(rich_texts, :trix) { |name| name.delete_prefix("rich_text_") }
+        discover_attachments
+        discover_basic_associations
+      end
+
+      private
+
+      # Fetches the model class, falling back to the items_holder parent record in certain instances (e.g. in the context of the sidebar)
+      def safe_model_class
+        respond_to?(:model_class) ? model_class : @items_holder.parent.record.class
+      rescue ActiveRecord::NoDatabaseError, ActiveRecord::ConnectionNotEstablished
+        nil
+      end
+
+      # Determines if a column is included in the discovery scope.
+      # A column is in scope if it's included in `only` and not in `except`.
+      def column_in_scope?(column_name)
+        (!@only || @only.include?(column_name)) && (!@except || !@except.include?(column_name))
+      end
+
+      def determine_field_config(attribute, column)
+        if safe_model_class.respond_to?(:defined_enums) && safe_model_class.defined_enums[attribute.to_s]
+          return { field: "select", enum: "::#{safe_model_class.name}.#{attribute.to_s.pluralize}" }
+        end
```

Comment:
Is it possible to optimize this check? Right now for each field we're checking if there is an enum